src/main/java/be/seeseemelk/mockbukkit/MockUnsafeValues.java

```diff
@@ -116,20 +117,29 @@ public void reportTimings()
 	@Override
 	public Material fromLegacy(Material material)
 	{
-		return material;
+		if (material == null || !material.isLegacy())
+		{
+			return material;
+		}
+		throw new UnimplementedOperationException();
```

Comment:
Why are still throwing an UnimplementedOperationException here?

Reply:
Because Bukkit has more logic there that we haven't implemented. This only implements one execution path.

Reply:
Why haven't we implemented that in this PR?

Reply:
It relies on NMS code that's not feasible for us to implement. You can look at `CraftLegacy`'s static initializer to see what I mean.